content/learn/07.built-in-libraries/02.i2s/i2s.md

```diff
@@ -113,31 +115,44 @@ The next sample of incoming I2S data available (or 0 if no data is available)
 ### `write()`
 
 #### Description
-Writes binary data to the I2S interface. This data is sent as a sample or series of samples.
+Writes binary data to the I2S interface. Data is sent either as one sample or a sequence of samples.
+
+The single argument function `write(val)` writes one sample for one channel. A full write requires
+calling this function twice, once for the left channel and once for the right channel.
+If the device buffer is full, the function blocks until the data can be written.
+Before writing, the value `val` will be converted to the appropriate number of bits per sample.
+
+The two-argument function `write(buf, len)` writes a sequence of samples from the array `buf`.
+Data must be stored in the proper format for the sample size, with the left and right channels
+interleaved, and the argument `len` specifies the number of **bytes** in the buffer, not the
+number of samples. If the device buffer is too full to hold all the data, the function writes
+as much as will fit (possibly nothing), and returns without blocking. The return value is the
+number of bytes that was actually written. It is the responsibility of the application to
+handle any such partial writes and act accordingly.
```

Comment:
Be more generic, always referring to the 2 channel scenario

Reply:
Agreed. my bad. See above.